kernel/kselftest.py

```diff
@@ -80,10 +80,27 @@ def setUp(self):
         if self.run_type == 'upstream':
             location = self.params.get('location', default='https://github.c'
                                        'om/torvalds/linux/archive/master.zip')
-            tarball = self.fetch_asset("kselftest.zip", locations=[location],
-                                       expire='1d')
-            archive.extract(tarball, self.workdir)
-            self.buldir = os.path.join(self.workdir, 'linux-master')
+            path = ''
+            git_location = False
+            kselfdir = ''
+            if ".zip" in location:
+                kselfdir = "kselftest.zip"
+                git_location = True
+            elif ".tar" in location:
+                kselfdir = "kselftest.tar"
+                git_location = True
+            if git_location:
```

Comment:
Just one last comment, the var name is confusing. it must be either git_location as `False` or var name needs to change.

Reply:
ok

Reply:
done, pushed the changes please have a look